File: app/src/main/java/com/example/umonshoraire/AmphiAdapter.java
package com.example.umonshoraire;

import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class AmphiAdapter extends RecyclerView.Adapter<AmphiAdapter.ViewHolder> {

    private final ArrayList<Amphi> listeAmphi;

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView summaryView;
        private final TextView dateStartView;
        private final TextView dateEndView;

        public ViewHolder(View itemView) {
            super(itemView);

            summaryView = (TextView) itemView.findViewById(R.id.summaryView);
            dateStartView = (TextView) itemView.findViewById(R.id.dateStartView);
            dateEndView = (TextView) itemView.findViewById(R.id.dateEndView);

        }

        public TextView getSummaryView() {
            return summaryView;
        }
        public TextView getDateStartView() { return dateStartView;}
        public TextView getDateEndView() { return dateEndView; }
    }

    public AmphiAdapter(ArrayList<Amphi> listeAmphi) {
        this.listeAmphi = listeAmphi;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.text_row_item, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.getSummaryView().setText(listeAmphi.get(position).summary);
        holder.getSummaryView().setTypeface(holder.getSummaryView().getTypeface(), Typeface.BOLD);
        holder.getDateStartView().setText(listeAmphi.get(position).getDateStart());
        holder.getDateEndView().setText(listeAmphi.get(position).getDateEnd());
    }

    @Override
    public int getItemCount() {
        return listeAmphi.size();
    }


}

File: app/src/main/java/com/example/umonshoraire/AmphiAdapter.kt
package com.example.umonshoraire

import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class AmphiAdapter() : RecyclerView.Adapter<AmphiAdapter.ViewHolder>() {

    constructor(listeAmphi: ArrayList<Amphi>) : this() {
        this.listAmphi = listeAmphi
    }

    private lateinit var listAmphi : ArrayList<Amphi>

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {


        var summaryView: TextView
        var dateStartView: TextView
        var dateEndView: TextView

        init {
            summaryView = itemView.findViewById(R.id.summaryView)
            dateStartView = itemView.findViewById(R.id.dateStartView)
            dateEndView = itemView.findViewById(R.id.dateEndView)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.text_row_item, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return listAmphi.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.summaryView.text = listAmphi[position].summary
        holder.summaryView.setTypeface(holder.summaryView.typeface, Typeface.BOLD)
        holder.dateStartView.text = listAmphi[position].getDateStart()
        holder.dateEndView.text = listAmphi[position].getDateEnd()
    }

}